pybot/endpoints/slack/utils/event_utils.py

```diff
@@ -103,3 +106,69 @@ def build_messages(user_id) -> Tuple[Message, Message, Message, Message]:
         data = await response.json()
         headers = {"Authorization": f"Bearer {data['token']}"}
     return headers
+
+
+async def get_profile_suggestions(
+    slack_id: int,
+    auth_header: Dict[str, str],
+    slack_api: SlackAPI,
+    session: ClientSession,
+) -> List[Tuple[str, str]]:
+    """
+    Generates a list of slack channels to suggest
+
+    :param slack_id: The slack_id to return profile information from
+    :return: A list of slack channel names to suggest for the user
+    """
+    results = []
+
+    # Get profile data that we want to build suggestions from.
+    user_info = await slack_api.query(methods.USERS_INFO, {"user": slack_id})
+    email = user_info["user"]["profile"]["email"]
+
+    async with session.get(
+        f"{BACKEND_URL}/auth/profile/admin/",
+        headers=auth_header,
+        params={"email": email},
+    ) as response:
+        data = await response.json()
+        logger.info(f"Retrieved profile data from email: {email}")
+
+    # Get the names of all public, non-archived channels
+    channels = await slack_api.query(
+        methods.CONVERSATION_LIST, {"exclude_archived": True}
```

Comment:
I like how you excluded the archive, but since the api doesn't have the ability to limit private channels we can use the private state to exclude those:
https://api.slack.com/methods/channels.list

Reply:
The conversation list method excludes private channels by default, is this what you mean?
https://api.slack.com/methods/conversations.list

Reply:
ahh cool.